Clean up the key configuration

dj-bauer opened the pull request:
I know this project hasn't had any commits since 2013 and seems like a dead project for you but I just wanted to contribute something since I checked it out and did some quick changes.

Now all the key symbols are set in config.js and not hardcoded into the gameBoard.js and main.js
This could allow easier key-configuration later on.

dj-bauer committed:
Clean up the key configuration

Now all the key symbols are set in config.js and not hardcoded into the gameBoard.js and main.js
This could allow easier key-configuration later on.

dj-bauer committed:
Improve the item pickup function

Instead of manually specifying the function for the item pickup individually, move the important stupp into a separate function and just do the specific stuff in the if check

dj-bauer committed:
Fix the item pickup

Fixes an issue with the previously created pickup function, which deletes the item from the board before running the custom functions for it.

dj-bauer committed:
Transform the function objects of Tile and Item to proper javascript functions

dj-bauer committed:
Fix html issue with body tag not being closed

hpeinar commented on includes/js/gameBoard.js:
```diff
@@ -237,28 +237,28 @@ function gameBoard() {
 
 					// shift combinations
-					if(keyDownCode == 539) {
+					if(keyDownCode === (500+config.key_right)) {
```
As 539 keyCode doesn't exist, I assume 500 is "shiftModifier" which is just added to the keyCode when shift key is held down.

If you are already cleaning up and making things more readable, it would make sense to do either:
1. Have actual key events in the keyQueue so all the key press event information would be accessible here
2. Have "shiftModifier" as a config variable as well so the code would be understandable ie: `config.shiftModifier + config.keyRight`

dj-bauer replied:
That makes perfectly sense and I will add option 2

hpeinar commented on includes/js/config.js:
```diff
@@ -11,5 +11,12 @@ var config = {
+	key_right: 39,
+	key_down: 40,
+	key_use: 13,
+	key_shift: 16
```
All other configuration variables are in camelCase, it would be a good custom to keep the trend.
If you want more clarity, you could create keys: {} objects and list the keys under there.

hpeinar commented on includes/js/item.js:
```diff
@@ -4,20 +4,25 @@ http://projekt406.ee/codeblog
-	this.animation = false;
-	this.animationSpeed = 2;
-	this.draw = function(board, canvas) {
+class Item {
```
Why just make item into a class? It would make sense to have everything as a class then (player and board as well)

dj-bauer replied:
Oupsie, I meant to only include the first commit #71c355f in the PR and the following commits, where mostly for me to fiddle around with the code to figure things out. But you are right, just having item and tile as a class and player and board a function is inconsistent.
Depending on how you feel I could also reformat the other 2 classes or just revert them back to the original state.

dj-bauer committed:
Add more clarity to the key configuration to comply with changes requested
Moved all the keycodes into a separate object KEYCODES and mapped them in the config object.
Also I renamed the key-variables to use camelCase.
